Update Go docs (#9992)
getsentry/sentry-docs: docs/platforms/go/guides/fiber/index.mdx

---
title: Fiber
description: "Learn how to add Sentry instrumentation to programs using the Fiber package."
---

> This page shows how to add Sentry instrumentation to programs using the Fiber

Reviewer: ```suggestion
```

Already in the description which shows on the webpage now.

package.

> For a quick reference, there is a [complete
> example](https://github.com/getsentry/sentry-go/tree/master/_examples/fiber)
> available directly at the Go SDK source code repository.

Reviewer: ```suggestion
The Sentry Go SDK repository has a [complete
example](https://github.com/getsentry/sentry-go/tree/master/_examples/fiber)
of how to set up Sentry for a Fiber app. 
```

> [GoDoc-style API
> documentation](https://godoc.org/github.com/getsentry/sentry-go/fiber) is also
> available.

Reviewer: ```suggestion
For additional reference, see the [`sentryfiber` API
documentation](https://godoc.org/github.com/getsentry/sentry-go/fiber).
```

			})
		}
		return ctx.SendStatus(fiber.StatusOK)
	})

	if err := app.Listen(":3000"); err != nil {
		panic(err)
	}
```

## Configure

`sentryfiber` accepts a struct of `Options` that allows you to configure how the handler will behave.

Currently, it respects three options:

```go
// Repanic configures whether Sentry should repanic after recovery, in most cases it should be set to true,
// as fiber includes its own Recover middleware that handles http responses.
Repanic bool
// WaitForDelivery configures whether you want to block the request before moving forward with the response.
// Because Fiber's `Recover` handler doesn't restart the application,
// it's safe to either skip this option or set it to `false`.
WaitForDelivery bool
// Timeout for the event delivery requests.
Timeout time.Duration
```

## Usage

`sentryfiber` attaches an instance of `*sentry.Hub` (https://godoc.org/github.com/getsentry/sentry-go#Hub) to the request's context, which makes it available throughout the rest of the request's lifetime.
You can access it by using the `sentryfiber.GetHubFromContext()` method on the context itself in any of your proceeding middleware and routes.

> And it should be used instead of the global `sentry.CaptureMessage`, `sentry.CaptureException`, or any other calls, as it keeps the separation of data between the requests.

Reviewer: ```suggestion
This method should be used instead of the global `sentry.CaptureMessage`, `sentry.CaptureException`, or any other calls, as it maintains the separation of data between the requests.
```
